src/users/users.service.ts

```diff
@@ -53,6 +98,44 @@ export class UsersService {
     }
   }
 
+  async getTokensByUsername(userName: string): Promise<AuthToken[]> {
+    const user = await this.getUserByUsername(userName);
+    return user.authTokens;
+  }
+
+  async removeToken(userName: string, timestamp: number) {
+    const user = await this.getUserByUsername(userName);
+    const token = await this.authTokenRepository.findOne({
+      where: { createdAt: new Date(timestamp), user: user },
```

Comment:
Could it be possible that a user generates two tokens with the same timestamp?

Maybe the identifier should be unique and the token will be deleted based on its identifier.

Reply:
Well to be frank I based that on your implementation in the frontend repo. I thought you had good reason to build it that way.

Reply:
Okay, I built it without having any knowledge about the backend. Timestamps are okay IMO, if the backend ensures that they're unique.

Reply:
Maybe it would be best to just expose the id and use that for the deletion? What do you think @davidmehren?